crm-api/src/main/java/com/agilemonkeys/test/crm/server/resource/exception/HandlerExceptions.java

```diff
@@ -0,0 +1,44 @@
+package com.agilemonkeys.test.crm.server.resource.exception;
+
+import com.agilemonkeys.test.crm.commons.exception.EntityNotFoundCRMException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.time.LocalDateTime;
+
+@ControllerAdvice
+public class HandlerExceptions extends ResponseEntityExceptionHandler {
+
+    Logger log = LoggerFactory.getLogger(HandlerExceptions.class);
+
+    @ExceptionHandler(EntityNotFoundCRMException.class)
+    protected ResponseEntity<Object> handleEntityNotFound(EntityNotFoundCRMException ex) {
+        ExceptionResponse exceptionResponse = new ExceptionResponse(LocalDateTime.now(),
+                HttpStatus.NOT_FOUND,
+                "Customer with ID [" + ex.getIdCustomer() + "] not found",
```

Comment:
With my comment about the exception I meant if you look at he message It seems we are only dealing with Customers but instead you're using with every entity you have and for example if you don't find a user you're throwing the same exception.

Reply:
Ok, yes this message is not generic. The correct (for me) it´s a generic message "No results found for ID [ id ]". I like  Exception for operation but i don´t like for model